packages/model-viewer/src/features/ar.ts

```diff
@@ -217,28 +217,26 @@ configuration or device capabilities');
     async[$selectARMode]() {
       this[$arMode] = ARMode.NONE;
       if (this.ar) {
-        const arModes: ARMode[] = [];
-        this[$arModes].forEach((value) => {
-          arModes.push(value);
-        });
-
-        for (const value of arModes) {
-          if (value === 'webxr' && IS_WEBXR_AR_CANDIDATE && !isWebXRBlocked &&
-              await this[$renderer].arRenderer.supportsPresentation()) {
-            this[$arMode] = ARMode.WEBXR;
-            break;
-          } else if (
-              value === 'scene-viewer' && IS_SCENEVIEWER_CANDIDATE &&
-              !isSceneViewerBlocked) {
-            this[$arMode] = ARMode.SCENE_VIEWER;
-            break;
-          } else if (value === 'quick-look' && IS_AR_QUICKLOOK_CANDIDATE) {
-            this[$arMode] = ARMode.QUICK_LOOK;
-            break;
+        if (this.src != null) {
+          for (const value of this[$arModes]) {
+            if (value === 'webxr' && IS_WEBXR_AR_CANDIDATE && !isWebXRBlocked &&
+                await this[$renderer].arRenderer.supportsPresentation()) {
+              this[$arMode] = ARMode.WEBXR;
+              break;
+            } else if (
+                value === 'scene-viewer' && IS_SCENEVIEWER_CANDIDATE &&
+                !isSceneViewerBlocked) {
```

Comment:
Nit. Since we're touching this code, a better approach could be as below. Having the else part only makes the code harder to read and based on the logic we can simply have:
`
for (const value of arModes) {
          if (value === 'webxr' && IS_WEBXR_AR_CANDIDATE && !isWebXRBlocked &&
              await this[$renderer].arRenderer.supportsPresentation()) {
            this[$arMode] = ARMode.WEBXR;
            break;
          } 
          if (
              value === 'scene-viewer' && IS_SCENEVIEWER_CANDIDATE &&
              !isSceneViewerBlocked) {
            this[$arMode] = ARMode.SCENE_VIEWER;
            break;
          } 
          if (value === 'quick-look' && IS_AR_QUICKLOOK_CANDIDATE) {
            this[$arMode] = ARMode.QUICK_LOOK;
            break;
          }
        }`

Reply:
Good call, done.